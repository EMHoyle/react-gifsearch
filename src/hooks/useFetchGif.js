import { useState, useEffect } from "react";
import { getGif } from '../helpers/getGif';

const useFetchGif = (category) => {
  
    const [state, setState] = useState({
        data: [],
        loading: true,
    });

    useEffect(() => {
        
        getGif(category)
            .then(img => {
                setTimeout(() => {
                    setState({
                        data: img,
                        loading: false
                    });
                }, 500);
            })
    }, [category])

    return state;
}

export default useFetchGif
