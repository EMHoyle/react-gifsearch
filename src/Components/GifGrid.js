import React from 'react'
import useFetchGif from '../hooks/useFetchGif';
import GifGridItems from './GifGridItems';

const GifGrid = ({category}) => {

    const {data:images, loading} = useFetchGif(category);

    return (
        <>
        <h4>{category}</h4>
        { loading && <p className='animate__animated animate__flash'>Loading...</p>}

        <div className="card-grid">
                {
                    images.map( img => {
                        return (
                            <GifGridItems 
                                key = {img.id}
                                {...img}
                            />
                        )
                    })
                }
        </div>
        </>
    )
}

export default GifGrid;
