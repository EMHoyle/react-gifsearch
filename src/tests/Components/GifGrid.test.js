import React from 'react';
import '@testing-library/jest-dom';
import GifGrid from '../../Components/GifGrid';
import { shallow } from 'enzyme';
import useFetchGif from '../../hooks/useFetchGif';
jest.mock('../../hooks/useFetchGif');
 
describe('GifGrid Test', () => {
    const category = 'HIMYM';
    // const wrapper = shallow(<GifGrid category={category}/>);
 
    test('Test 1 - Enzyme: GifGrid should be render correctly', () => {
        
        useFetchGif.mockReturnValue({
            data: [],
            loading: true,
        });

        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot(); 
    });
 
    test('Test 2 - Enzyme: GifGrid should render elements when loading images with Fetch', () => {
        
        const gifs = [{
            id: 'ABC',
            url: 'https://google.com',
            title: 'ABC Title'
            
        }]
        
        useFetchGif.mockReturnValue({
            data: gifs,
            loading: false,
        });
        
        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItems').length).toBe( gifs.length );
 
    });
})