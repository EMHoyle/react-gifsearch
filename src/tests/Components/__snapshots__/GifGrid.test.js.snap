// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid Test Test 1 - Enzyme: GifGrid should be render correctly 1`] = `
<Fragment>
  <h4>
    HIMYM
  </h4>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`GifGrid Test Test 2 - Enzyme: GifGrid should render elements when loading images with Fetch 1`] = `
<Fragment>
  <h4>
    HIMYM
  </h4>
  <div
    className="card-grid"
  >
    <GifGridItems
      id="ABC"
      key="ABC"
      title="ABC Title"
      url="https://google.com"
    />
  </div>
</Fragment>
`;
